packages/agreement-platformstate-writer/test/consumerServiceV2.integration.test.ts

```diff
@@ -0,0 +1,2035 @@
+        clientId: generateId(),
+        kid: `kid ${Math.random()}`,
+        purposeId: generateId(),
+      });
+      const previousTokenStateEntry2: TokenGenerationStatesClientPurposeEntry =
+        {
+          ...getMockTokenStatesClientPurposeEntry(tokenStateEntryPK2),
+          agreementState: itemState.inactive,
+          GSIPK_consumerId_eserviceId,
+          GSIPK_eserviceId_descriptorId: undefined,
+          descriptorAudience: undefined,
+          descriptorState: undefined,
+        };
+      await writeTokenStateEntry(previousTokenStateEntry2, dynamoDBClient);
+
+      await sleep(1000, mockDate);
+      await handleMessageV2(message, dynamoDBClient);
+
+      // platform-states
+      const retrievedAgreementEntry = await readAgreementEntry(
+        agreementEntryPrimaryKey,
+        dynamoDBClient
+      );
+
+      const expectedAgreementEntry: PlatformStatesAgreementEntry = {
+        PK: agreementEntryPrimaryKey,
+        version: 1,
+        state: itemState.active,
+        updatedAt: agreement.stamps.activation!.when.toISOString(),
+        GSIPK_consumerId_eserviceId: makeGSIPKConsumerIdEServiceId({
+          consumerId: agreement.consumerId,
+          eserviceId: agreement.eserviceId,
+        }),
+        GSISK_agreementTimestamp:
+          agreement.stamps.activation!.when.toISOString(),
+        agreementDescriptorId: agreement.descriptorId,
+      };
+      expect(retrievedAgreementEntry).toEqual(expectedAgreementEntry);
+
+      // token-generation-states
+      const retrievedTokenStateEntries =
+        await readTokenStateEntriesByConsumerIdEserviceId(
+          GSIPK_consumerId_eserviceId,
+          dynamoDBClient
+        );
+      const expectedTokenStateEntry1: TokenGenerationStatesClientPurposeEntry =
+        {
+          ...previousTokenStateEntry1,
+          agreementState: itemState.active,
+          descriptorState: catalogEntry.state,
+          descriptorAudience: catalogEntry.descriptorAudience,
+          descriptorVoucherLifespan: catalogEntry.descriptorVoucherLifespan,
+          updatedAt: new Date().toISOString(),
+        };
+      const expectedTokenStateEntry2: TokenGenerationStatesClientPurposeEntry =
+        {
+          ...previousTokenStateEntry2,
+          agreementState: itemState.active,
+          descriptorState: catalogEntry.state,
+          descriptorAudience: catalogEntry.descriptorAudience,
+          descriptorVoucherLifespan: catalogEntry.descriptorVoucherLifespan,
+          updatedAt: new Date().toISOString(),
+        };
+
+      expect(retrievedTokenStateEntries).toHaveLength(2);
+      expect(retrievedTokenStateEntries).toEqual(
+        expect.arrayContaining([
+          expectedTokenStateEntry1,
+          expectedTokenStateEntry2,
+        ])
+      );
+    });
+  });
+  describe("AgreementSuspendedByProducer, AgreementSuspendedByConsumer, AgreementSuspendedByPlatform", async () => {
```

Comment:
it's better to loop over the event name and use it for each test, otherwise we are testing just one of them

Reply:
It required some verbose logic inside the tests. We decided to almost replicate all the tests (even more verbose, but faster)